Lib/MaybeBool.hpp

```diff
@@ -37,6 +37,7 @@ class MaybeBool
   MaybeBool(Value val) : _value(val) {}
 
   bool known() const { return _value!=UNKNOWN; }
+  bool maybe() const { return _value==UNKNOWN; }
```

Comment:
Should we maybe rename that to `unknown`? In order to be consistent with the names of the enum variants.

Reply:
`maybe` makes more sense semantically than `unknown`.  If  `isConstraintTerm(t).maybe()` suggests that `t` could possibly be a constraint term, whilst `isConstraintTerm(t).unknown()` just says that I don't know whether `t` is or is not.

I would suggest changing the names of the enum variants.